src/Core/Framework/Test/Api/Controller/InfoControllerTest.php

```diff
@@ -142,4 +152,63 @@ public function testBusinessEventRoute(): void
             static::assertEquals($event, $actualEvents[0]);
         }
     }
+
+    public function testBundlePaths(): void
+    {
+        $kernelMock = $this->createMock(Kernel::class);
+        $packagesMock = $this->createMock(Packages::class);
+        $infoController = new InfoController(
+            $this->createMock(DefinitionService::class),
+            new ParameterBag([
+                'kernel.shopware_version' => 'shopware-version',
+                'kernel.shopware_version_revision' => 'shopware-version-revision',
+                'shopware.admin_worker.enable_admin_worker' => 'enable-admin-worker',
+                'shopware.admin_worker.transports' => 'transports',
+            ]),
+            $this->createMock(BusinessEventRegistry::class),
+            $kernelMock,
+            $packagesMock,
+            $this->createMock(BusinessEventCollector::class),
+            true,
+            []
+        );
+
+        $infoController->setContainer($this->createMock(Container::class));
+
+        $assetPackage = $this->createMock(Package::class);
+        $packagesMock
+            ->expects(static::exactly(1))
+            ->method('getPackage')
+            ->willReturn($assetPackage);
+        $assetPackage
+            ->expects(static::exactly(1))
+            ->method('getUrl')
+            ->willReturnArgument(0);
+
+        $filesystemRoot = vfsStream::setup();
+        vfsStream::create([
+            'Resources' => [
+                'public' => [
+                    'administration' => [
+                        'js' => [
+                            'some-functionality-bundle.js' => '// File contents',
+                        ],
+                    ],
+                ],
+            ],
+        ], $filesystemRoot);
+        $kernelMock
+            ->expects(static::exactly(1))
+            ->method('getBundles')
+            ->willReturn([new BundleFixture('SomeFunctionalityBundle', $filesystemRoot->url())]);
+
+        $config = json_decode($infoController->config()->getContent(), true);
+        static::assertArrayHasKey('SomeFunctionalityBundle', $config['bundles']);
+
+        $jsFilePath = explode('?', $config['bundles']['SomeFunctionalityBundle']['js'][0])[0];
+        static::assertEquals(
+            'bundles/somefunctionality/administration/js/some-functionality-bundle.js',
+            $jsFilePath
+        );
```

Comment:
Hi, great idea to abstract away the filesystem 👍🏼 We already have a library doing that in the core though, namely Flysystem. This also offers a fast in-memory adapter for usage in tests. We'd rather not include a new dependency for this, so please consider one of these alternatives:

1. See my suggestion below (quickly thrown together, so better don't take it verbatim 😅) and use the `Local` adapter with a real but temporary file even though it is not as performant asa virtual filesystem
2. Improve the testability of the `InfoController`, for example by replacing the two native `file_exists` calls with Flysystem calls. Then the flysystem adapter could be swapped with the in-memory one [already available](https://github.com/shopware/development/blob/master/composer.json#L52) in the core dependencies for testing

```suggestion
        $rootPath = implode(DIRECTORY_SEPARATOR, [
            sys_get_temp_dir(),
            Uuid::randomHex()
        ]);
        $filePath = 'Resources/public/administration/js/some-functionality-bundle.js';

        $filesystem = new Filesystem(new Local($rootPath));
        $filesystem->write($filePath, '// File contents');

        static::assertTrue($filesystem->has($filePath));

        $kernelMock
            ->expects(static::exactly(1))
            ->method('getBundles')
            ->willReturn([new BundleFixture('SomeFunctionalityBundle', $rootPath)]);

        $config = json_decode($infoController->config()->getContent(), true);
        static::assertArrayHasKey('SomeFunctionalityBundle', $config['bundles']);

        $jsFilePath = explode('?', $config['bundles']['SomeFunctionalityBundle']['js'][0])[0];
        static::assertEquals(
            'bundles/somefunctionality/administration/js/some-functionality-bundle.js',
            $jsFilePath
        );

        $filesystem->getAdapter()->deleteDir('/');

        static::assertFalse($filesystem->has($filePath));
```